silkworm/core/execution/call_tracer.cpp

```diff
@@ -38,25 +85,33 @@ void CallTracer::on_execution_start(evmc_revision /*rev*/, const evmc_message& m
     }
 }
 
-void CallTracer::on_instruction_start(uint32_t pc, const intx::uint256* stack_top, int stack_height, int64_t /*gas*/,
+void CallTracer::on_instruction_start(uint32_t pc, const intx::uint256* stack_top, int stack_height, int64_t gas,
                                       const evmone::ExecutionState& state, const IntraBlockState& intra_block_state) noexcept {
     const auto op_code = state.original_code[pc];
     if (op_code == evmc_opcode::OP_CREATE) {
+        const auto& cost_table{get_baseline_cost_table(state.rev, state.analysis.baseline->eof_header.version)};
+        if (const auto status{check_requirements<Opcode::OP_CREATE>(cost_table, gas, stack_top, stack_top - stack_height)}; status != EVMC_SUCCESS) {
+            return;  // Early failure in pre-execution checks, do not trace anything for compatibility w/ Erigon
+        }
```

Comment:
Duplicated code, can we move `cost_table` and `status` to check_requirements? Then we would have something like:
```
if (!requirements_checked<Opcode::OP_CREATE>(state, gas, stack_top, stack_heightd) {
    return; // Early failure in pre-execution checks, do not trace anything for compatibility w/ Erigon
}
```